tests/test_views.py

```diff
@@ -0,0 +1,168 @@
+from unittest.mock import patch
+import unittest
+import report_app
+from moto import mock_dynamodb
+import boto3
+
+
+class TestFunctionalViews(unittest.TestCase):
+    def setUp(self):
+        app = report_app.app
+        app.config["TESTING"] = True
+        self.ctx = app.app_context()
+        self.ctx.push()
+        self.client = app.test_client()
+        self.index = "/index"
+
+    def tearDown(self):
+        self.ctx.pop()
+
+    def test_index(self):
+        assert self.client.get(self.index).status_code == 200
+
+    def test_default(self):
+        assert self.client.get("/").status_code == 200
+
+    def test_home_with_no_auth(self):
+        assert self.client.get("/home").status_code == 302
+        assert self.client.get("/home").headers.get("location") == self.index
+
+    def test_start_with_no_auth(self):
+        assert self.client.get("/start").status_code == 302
+        assert self.client.get("/start").headers.get("location") == self.index
+
+    def test_callback_server_error(self):
+        assert self.client.get("/callback").status_code == 500
+
+
+@patch.dict('os.environ', {'DYNAMODB_TABLE_NAME': 'TEST_TABLE'})
+@patch.dict('os.environ', {'DYNAMODB_REGION': 'eu-west-2'})
+@patch.dict('os.environ', {'DYNAMODB_ACCESS_KEY_ID': 'FAKE'})
+@patch.dict('os.environ', {'DYNAMODB_SECRET_ACCESS_KEY': 'FAKE'})
+class TestGitHubReports(unittest.TestCase):
```

Comment:
Love the tests 💖

We should probably add some tests to ensure the deprecated endpoints are still callable - doesn't look like the tests existed before though so happy for it to be left🧪